File: Java/Medium/Find-the-Duplicate-Number.java
class Solution {
    public int findDuplicate(int[] nums) {
        boolean[] count = new boolean[nums.length];
        for (int num:nums) {
            if (count[num]) {
                return num;
            } else {
                count[num]=true;
            }
        }
        return nums[0];
    }
}

File: Python/Medium/Find-the-Duplicate-Number.py
class Solution(object):
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        numDict = [False for x in range(len(nums))]
        for num in nums:
            if (numDict[num] is True):
                return num
            else:
                numDict[num] = True
        return nums[0]
        
